File: backend/src/main/java/com/anmory/yunji/entity/Contraction.java
package com.anmory.yunji.entity;

import lombok.Data;

import java.time.LocalDateTime;

/**
 * 宫缩记录
 */
@Data
public class Contraction {
    private Integer contractionId;
    private Integer userId;
    private LocalDateTime startedAt;
    private Integer durationSeconds;
    private LocalDateTime createdAt;
}

File: frontend/lib/api/contraction.ts
import { apiGet, apiPost, apiDelete } from "@/lib/api"

export interface ContractionRecord {
  contractionId: number
  userId: number
  startedAt: string
  durationSeconds: number
  createdAt?: string
}

export async function addContraction(userId: number, startedAt: string, durationSeconds: number) {
  return apiPost<ContractionRecord>("/api/contraction/add", { userId, startedAt, durationSeconds })
}

export async function getContractionsByDate(userId: number, date: string) {
  return apiGet<ContractionRecord[]>("/api/contraction/list", { userId, date })
}

export async function clearContractionsByDate(userId: number, date: string) {
  return apiDelete<void>("/api/contraction/clear", { userId, date })
}
